### PR TITLE
Feature/col 577 default layout

### PR DESCRIPTION
**Pull request by ayang99**

### CODE REVIEW

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 1]
```diff
@@ -132,39 +134,47 @@ public int execute() throws Exception {
...
     terminal.println("");
-    for (val entity : filterEntities(entities)) {
+
+    // filtering based on file name only applies to FILENAME and BUNDLE layouts
+    Set<Entity> entitySet = entities;
+    if (layout != OutputLayout.ID) {
+      entitySet = filterEntities(entities);
+    }
+
+    // sweepExisting(entitySet);
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Remove?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 2]
```diff
@@ -132,39 +134,47 @@ public int execute() throws Exception {
...
 
     if (!verifyLocalAvailableSpace(entities)) {
       return FAILURE_STATUS;
     }
 
     int i = 1;
     terminal.println("");
-    for (val entity : filterEntities(entities)) {
+
+    // filtering based on file name only applies to FILENAME and BUNDLE layouts
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Caps

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 3]
```diff
@@ -235,7 +248,8 @@ private File getLayoutTarget(Entity entity) {
    */
   private Set<Entity> resolveEntities(List<String> objectIds) {
     // Set to remove duplicates
-    val entities = ImmutableSet.<Entity> builder();
+    // we don't want it to be immutable (there's going to be subsequent filtering)
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Caps

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 4]
```diff
@@ -248,26 +262,44 @@ private File getLayoutTarget(Entity entity) {
       }
     }
 
-    return entities.build();
+    return entities;
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]:
```java
objectIds.stream().map(metadataService::getEntity).collect(toList())
```

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 5]
```diff
@@ -248,26 +262,44 @@ private File getLayoutTarget(Entity entity) {
...
-    val filtered = ImmutableSet.<Entity> builder();
-    for (val entity : entities) {
-      val file = getLayoutTarget(entity);
-      val skip = file.exists() && !force;
-      if (skip) {
-        continue;
-      }
 
-      filtered.add(entity);
+    // ImmutableListMultimap preserves insert order
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: 👍

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 6]
```diff
@@ -248,26 +262,44 @@ private File getLayoutTarget(Entity entity) {
...
+    val groupEntities = Multimaps.index(entities, Entity::getFileName);
+
+    val normalizedEntities = new HashSet<Entity>();
+    for (val fileName : groupEntities.keySet()) {
+      // we only process first entity encountered for each file name
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Caps

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 7]
```diff
@@ -248,26 +262,44 @@ private File getLayoutTarget(Entity entity) {
...
+      // contract for Multimap states if we have a key (file name), we have a collection with at least one member
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: All of these caps :)

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java [hunk 8]
```diff
@@ -132,39 +134,47 @@ public int execute() throws Exception {
...
     return SUCCESS_STATUS;
   }
 
   /**
-   * Prepares the local file system for moving files into after they have completed downloading.
+   * Checks local output directory for pre-existing files (before downloading them). Modifies the collection passed in.
    */
   @SneakyThrows
-  private void prepareLayout(Set<Entity> entities) {
+  private void sweepExisting(Set<Entity> entities) {
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/command/DownloadCommand.java]: Agreed

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/config/ClientBanner.java [hunk 9]
```diff
@@ -97,7 +97,11 @@ private static void log(StandardEnvironment env) {
       if (source instanceof EnumerablePropertySource) {
         val enumerable = (EnumerablePropertySource<?>) source;
         for (val propertyName : Sets.newTreeSet(ImmutableSet.copyOf(enumerable.getPropertyNames()))) {
-          log.info("            - {}: {}", propertyName, enumerable.getProperty(propertyName));
+          if (propertyName.equalsIgnoreCase("accessToken")) {
+			// suppress output
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/config/ClientBanner.java]: Formatting of indent, caps

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/Formats.java [hunk 10]
```diff
@@ -20,7 +20,7 @@
 public class Formats {
 
   public static String formatBytes(long bytes) {
-    int unit = 1000;
+    int unit = 1024;
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/Formats.java]: Huh. Was this affecting reporting?

Comment (ayang99) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/Formats.java]: units reported as KB/MB but unit = 1000 is actually KiB/MiB. Or the other way around.

**Pull request merged**